console/src/utils/version.ts

```diff
@@ -0,0 +1,29 @@
+import semver from "semver";
+
+const VERSION_REGEX = /^\d+\.\d+\.\d+$/g;
+
+/**
+ * Checks if a given version satisfies a required version range.
+ *
+ * @see https://github.com/halo-dev/halo/blob/a5a69780a37410d2734043d6cae1b718b57c722b/application/src/main/java/run/halo/app/infra/utils/VersionUtils.java#L18
+ * @param version - The version to check.
+ * @param requires - The required version range.
+ * @returns A boolean indicating whether the version satisfies the required range.
+ */
+export function satisfiesRequires(version?: string, requires?: string): boolean {
+  if (!version) {
+    version = "0.0.0";
+  }
+
+  if (!requires) {
+    return true;
+  }
+
+  requires = requires.trim();
+
+  if (VERSION_REGEX.test(requires)) {
+    requires = `>=${requires}`;
+  }
+
+  return version === "0.0.0" || semver.satisfies(version, requires, { includePrerelease: true });
```

Comment:
```suggestion
  return semver.satisfies(version, requires, { includePrerelease: true });
```
version 不存在只有可能在开发模式的情况下出现,当不存在时已经赋值为 0.0.0 所以直接走 semver.satisfies 即可,而当生产模式的时候不会存在 version 不存在的情况所以将 version === "0.0.0" 去掉可以在每次检查时简化一次判断

Reply:
用 idea 运行的时候，`/actuator/info` 就没有提供版本号。

<img width="1672" alt="image" src="https://github.com/halo-dev/plugin-app-store/assets/21301288/633a20c3-9f66-42cd-b29d-7635e933e50d">

Reply:
> 用 idea 运行的时候，`/actuator/info` 就没有提供版本号。

这种情况就是说的开发的时候, 在服务器是没有问题的,开发的时候直接走 semver.satisfies 即可不用考虑有没有多判断,但是在生产模式的时候可以不用判断,或者是 version 没有时直接返回 true 也行